great_expectations/data_context/data_context.py

```diff
@@ -845,6 +845,40 @@ def _load_config_variables_file(self):
         else:
             return {}
 
+    def get_config_with_variables_substituted_from_yaml_str(self, yaml_config: str):
```

Comment:
FYI - I'm breaking this out into a separate function since I think we may use this for our `add_datasource()` method as well, and now it is easier to test.